src/metabase/api/dashboard.clj

```diff
@@ -820,6 +823,26 @@
              field             (param->fields param)]
          (assoc field :value value))))
 
+(defn filter-values-from-field-refs
+  "Get filter values when only field-refs (e.g. `[:field \"SOURCE\" {:base-type :type/Text}]`)
+  are provided (rather than field-ids). This is a common case for nested queries."
+  [dashboard param-key]
+  (let [dashboard       (t2/hydrate dashboard :resolved-params)
```

Comment:
I think you don't need to hydrate the dashboard here, since it's passed in hydrated already.